packages/datacommons-mcp/datacommons_mcp/clients.py

```diff
@@ -344,6 +354,201 @@ async def _search_indicators(
+            # Return empty results on failure
+            return SearchResult(), {}
+
+        # Apply existence filtering if places are specified
+        if all_place_dcids:
+            # Ensure place variables are cached for all places in parallel
+            await asyncio.gather(
+                *(
+                    asyncio.to_thread(self._ensure_place_variables_cached, place_dcid)
+                    for place_dcid in all_place_dcids
+                )
+            )
+
+            # Filter topics and variables by existence (OR logic)
+            search_result.topics = self._filter_topics_by_existence_new(
+                search_result.topics, all_place_dcids
+            )
+            search_result.variables = self._filter_variables_by_existence_new(
+                search_result.variables, all_place_dcids
+            )
+
+        # Get member information for topics
+        if search_result.topics:
+            self._get_topics_members_with_existence_new(
+                search_result.topics, all_place_dcids
+            )
+
+        # After filtering, create a set of all remaining DCIDs
+        final_dcids = set(search_result.topics.keys()) | set(
+            search_result.variables.keys()
+        )
+
+        # Filter the original dcid_name_mappings to only include the final DCIDs
+        final_dcid_name_mappings = {
+            dcid: name
+            for dcid, name in dcid_name_mappings.items()
+            if dcid in final_dcids
+        }
+        return search_result, final_dcid_name_mappings
+
+    def _transform_search_indicators_response(
+        self, api_response: dict
+    ) -> tuple[SearchResult, dict]:
+        """
+        Transforms the response from the /api/nl/search-indicators endpoint.
+
+        Args:
+            api_response: The JSON response from the search-indicators API.
+
+        Returns:
+            A tuple containing a SearchResult object and a dcid_name_mappings dictionary.
+
+        Example Output:
+            (
+                SearchResult(
+                    topics={"dc/topic/Health": SearchTopic(dcid="dc/topic/Health", ...)},
+                    variables={"Count_Person": SearchVariable(dcid="Count_Person", ...)}
+                ),
+                {"dc/topic/Health": "Health", "Count_Person": "Person Count"}
+            )
+        """
+        unique_indicators = {}
+        query_results = api_response.get("queryResults", [])
+
+        # Iterate indicators from each query + index result and merge into single dict
+        for query_result in query_results:
+            for index_result in query_result.get("indexResults", []):
+                for indicator in index_result.get("results", []):
+                    dcid = indicator.get("dcid")
+                    # Deduplicate based on DCID, keeping the first one found.
+                    if dcid and indicator.get("name") and dcid not in unique_indicators:
+                        unique_indicators[dcid] = indicator
```

Comment:
This dict will respect insertion order. In my testing, the first index was `user_all_minilm_mem` with our custom data and the second `medium_ft` (not sure how you determine the order).

Since each index can return 20 results, given the result limit parameter, the results from the second index (or beyond) will always get dropped. 

Perhaps a good alternative would be to keep track of the cosine scores a bit longer and sort the search result which gets returned in 492 by that score so that when the truncation happen you get the highest X results rather than the highest X results from the first index.

Reply:
@clincoln8 just wanted to make sure this didn't get lost given the different threads. It won't impact us for now as we'll only do custom, but would have an impact if we ever did mixed